### DISCUSSION TITLE
SSR support

### DISCUSSION THREAD

**Discussion by Gounlaf** [tailflow/laravel-orion-ts#48] [Ideas]
Hi,

I've just discovered your library and I really think It worth to try it :)

I'm facing issue with Next.js compatibility. So I quickly tried with a CRA, it does work without any problem

```js
import logo from './logo.svg';
import './App.css';

import {Orion} from "@tailflow/laravel-orion/lib/orion";

Orion.init('https://your-api.test');
Orion.setToken('access-token-here');

function App() {
  console.debug(Orion.getToken())// Will output 'access-token-here' as expected

  return (
    <div className="App">[...]</div>
  );
}

export default App;
```

With a Next.js, I got the error `SyntaxError: Cannot use import statement outside a module`.

Some related issues:
* https://github.com/asyncapi/asyncapi-react/issues/177
* https://github.com/vercel/next.js/issues/9890
* https://github.com/kylewetton/deviceful/issues/4

Digging deeper, I found this: https://stackoverflow.com/questions/62422174/cannot-use-import-statement-outside-a-module-with-next-js (answer: https://stackoverflow.com/a/62423865/2098861) ;
Related issue, if I understand it correctly:
* https://github.com/vercel/next.js/issues/9607

Could you take a look and make your library compatible both server&client side (if it's not really the case)?

In the meantime, I'm trying the "dynamic" imports or "useEffect" with import inside (found in related issues linked above).

I don't know how to help more, but ask if you need more details / help / reproducible example...

Thanks,

Regards.

Edit: looking further, it seems that the problem is more on next.js side than on libraries...

**Comment by alexzarbn** [comment]
Hi @Gounlaf,

Thank you for raising this issue. Indeed, SSR is something I have totally missed! The goal of this SDK is to provide a complete solution for consuming Orion-powered REST APIs, whether it be from a client, or a server side. 

That being said, I will look into providing support for SSR in the next releases ✨
